packages/app/src/systems/Network/services/network.ts

```diff
@@ -148,11 +149,22 @@ export class NetworkService {
     return provider.fetchNode();
   }
 
-  static async assertAddNetwork(input: NetworkInputs['addNetwork']) {
+  static async validateAddNetwork(input: NetworkInputs['addNetwork']) {
     const { name, url } = input.data;
     if (!isValidNetworkUrl(url)) {
       throw new Error('Invalid network URL');
     }
+    let nodeInfo;
+    try {
+      const provider = await Provider.create(url);
+      nodeInfo = await provider.fetchNode();
+    } catch (err: unknown) {
+      throw new Error('Node version is not compatible with the wallet.');
+    }
+    const nodeVersion = parseInt(nodeInfo.nodeVersion.replace('.', ''));
+    if (nodeVersion < MIN_NODE_VERSION) {
+      throw new Error('Node version is not compatible with the wallet.');
+    }
```

Comment:
We can use `compare-versions` to compare here, them we can have only the string on the config also, `0.22.0`.